addons/controllers/certificates_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"time"
+
+	adminregv1 "k8s.io/api/admissionregistration/v1"
+	cert2 "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	"knative.dev/pkg/webhook/certificates/resources"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/webhooks"
+)
+
+var _ = Describe("Webhook", func() {
+
+	Context("server's certificate and key", func() {
+		It("should be generated and written to the webhook server CertDir", func() {
+			secret, err := webhooks.UpdateTLSSecret(ctx, webhookScrtName, webhookServiceName, certPath, keyPath, addonNamespace)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(secret).NotTo(BeNil())
+			cert, err := cert2.CertsFromFile(certPath)
+			Expect(err).ToNot(HaveOccurred())
+			certPEM, err := cert2.EncodeCertificates(cert[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(certPEM).To(Equal(secret.Data[resources.ServerCert]))
+			key, err := keyutil.PrivateKeyFromFile(keyPath)
+			Expect(err).ToNot(HaveOccurred())
+			orgKey, err := keyutil.ParsePrivateKeyPEM(secret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+		})
+		It("should become invalid after one week", func() {
+			secret, err := webhooks.UpdateTLSSecret(ctx, webhookScrtName, webhookServiceName, certPath, keyPath, addonNamespace)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(secret).NotTo(BeNil())
+			err = webhooks.ValidateCertificate(secret, time.Hour*24) // valid cert life is one week. One day should not make it invalid
+			Expect(err).ShouldNot(HaveOccurred())
+			err = webhooks.ValidateCertificate(secret, 8*time.Hour*24) // in 8 days certificate should be  invalid
+			Expect(err).Should(HaveOccurred())
+
+		})
+	})
+
+	Context("Mutating and Validating Configurations", func() {
+		It("should be updated with CA bundle ", func() {
+			var err error
+			orgvwcfgs := &adminregv1.ValidatingWebhookConfigurationList{}
+			err = k8sClient.List(ctx, orgvwcfgs)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(orgvwcfgs.Items).ToNot(BeEmpty())
+
+			orgmwcfgs := &adminregv1.MutatingWebhookConfigurationList{}
+			err = k8sClient.List(ctx, orgmwcfgs)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(orgmwcfgs.Items).ToNot(BeEmpty())
+
+			secret, err := webhooks.UpdateCertificates(ctx, k8sConfig, certPath, keyPath, webhookScrtName, addonNamespace, webhookServiceName, "")
```

Comment:
Question: Do we assume we will set certain label for the webhookConfigurations that we create?

Reply:
yes